src/main/kotlin/org/rust/ide/intentions/JoinParameterListIntention.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsValueParameter
+import org.rust.lang.core.psi.RsValueParameterList
+import org.rust.lang.core.psi.ext.RsElement
+
+abstract class JoinListIntention<TList : RsElement, TElement : RsElement>(
+    listClass: Class<TList>,
+    elementClass: Class<TElement>,
+    intentionText: String
+) : ChopListIntention<TList, TElement>(listClass, elementClass, intentionText) {
```

Comment:
I find this inheritance weird (I know that kotlin has similar class hierarchy)
I'd prefer to create some parent class for both ("join" and "chop") subhierarchy, move all base properties and methods to it and make own `invoke` implementation for each subhierarchy instead of overriding
I.e.
```
                SomeCoolNameForParentClassListIntention
                       /               \
           ChopListIntention       JoinListIntention
                     /                    \ 
    ChopParameterListIntention    JoinParameterListIntention